src/containers/FittableContainer/FittableContainer.jsx

```diff
@@ -157,6 +157,20 @@ const FittableContainer = React.createClass({
       facultyGrid: facultyGrid,
     }
 
+    let weekParity = null
+    let weekNum = 0
+    let weekType = null
+
+    if ('periods' in this.props.semester) {
+      const period = periodFromDate(this.props.viewDate, this.props.semester.periods)
+
+      if (period) {
+        weekParity = periodWeekParity(this.props.viewDate, this.props.semester)
+        weekNum = semesterWeek(this.props.viewDate, this.props.semester)
+        weekType = period.type
+      }
+    }
```

Comment:
Why is this here in the container? This bunch of code maps semester and viewDate into parity, number and type. It should be isolated in a function (`Semester → Date → (Parity | null, Integer | 0, String | null)`).

Then you’re passing weekParity, weekNum and weekType through the Control component to the ViewDate component. It doesn’t make any sense to me. It should be in the ViewDate component, or in the semester module.